app/models/concerns/mobile_workflow/displayable.rb

```diff
@@ -1,163 +1,29 @@
-      validate_on_success!(on_success)
-      validate_button_style!(style)
-      
-      {type: :button, label: label, url: url, method: method, style: style, onSuccess: on_success, sfSymbolName: 'trash', materialIconName: 'delete'}.compact
-    end
-        
-    def mw_display_url_button(label:, url:, method: :put, style: :primary, confirm_title: nil, confirm_text: nil, on_success: :reload, sf_symbol_name: nil, material_icon_name: nil)
-      validate_on_success!(on_success)
-      validate_button_style!(style)
-      
-      {type: :button, label: label, url: url, method: method, style: style, confirmTitle: confirm_title, confirmText: confirm_text, onSuccess: on_success, sfSymbolName: sf_symbol_name, materialIconName: material_icon_name}.compact
-    end
-    alias_method :mw_display_button_for_url, :mw_display_url_button
-    
-    def mw_display_system_url_button(label:, apple_system_url: nil, android_deep_link: nil, style: :primary, sf_symbol_name: nil, material_icon_name: nil)
-      validate_button_style!(style)
-      raise 'Invalid android_deep_link' if android_deep_link && !android_deep_link.start_with?('http')
-      
-      {type: :button, label: label, appleSystemURL: apple_system_url, androidDeepLink: android_deep_link, style: style, sfSymbolName: sf_symbol_name, materialIconName: material_icon_name}.compact
-    end
-    alias_method :mw_display_button_for_system_url, :mw_display_system_url_button
-    
-    def mw_display_modal_workflow_button(label:, modal_workflow_name:, style: :primary, on_success: :none, sf_symbol_name: nil, material_icon_name: nil)
-      validate_on_success!(on_success)
-      validate_button_style!(style)
-      
-      {type: :button, label: label, modalWorkflow: modal_workflow_name, style: style, onSuccess: on_success, sfSymbolName: sf_symbol_name, materialIconName: material_icon_name}.compact
-    end
-    alias_method :mw_display_button_for_modal_workflow, :mw_display_modal_workflow_button
-    
-    def mw_text_choice_question(question:, style:, text_choices:)
-      raise 'Missing question' if question.blank?
-      raise 'Text Choices should be a hash' unless text_choices.is_a?(Hash)
-      validate_question_style!(style)
-      
-      text_choices_a = text_choices.map{|k, v| {_class: "ORKTextChoice", exclusive: false, text: k, value: v} }.to_a
-      { question: question, answerFormat: { _class: "ORKTextChoiceAnswerFormat", style: style.to_s.camelize(:lower), textChoices: text_choices_a}}
-    end
-    
-    def mw_grid_large_section(id:, text:)
-      raise 'Missing id' if id.nil?
-      raise 'Missing text' if text.nil?
-      
-      { id: id, text: text, type: :largeSection }
-    end
-    
-    def mw_grid_small_section(id:, text:)
-      raise 'Missing id' if id.nil?
-      raise 'Missing text' if text.nil?
-      
-      { id: id, text: text, type: :smallSection }
+    extend ActiveSupport::Concern if defined?(Rails)
+
+    def self.included(base)
+      base.extend(Steps::Form)
     end
```

Comment:
I'm not sure if `self.included(base)` behaves the same when `defined?(Rails)` is `false` (and `extend ActiveSupport::Concern` is not inserted). Does it work properly when including `Displayable` in a plain Ruby class?

Reply:
I'm not sure if it's working because tests are requiring `rails_helper` which loads Rails, so apparently it's not testing the case where `Rails` is not defined.

Reply:
You could try replacing `require 'rails_helper'` with `require 'spec_helper'`in all `Displayable` tests to prevent loading Rails and see if it works. 
If it works, I think you could remove `extend ActiveSupport::Concern`. 
If it does not work, you could try to rewrite in a way that `ActiveSupport::Concern` is not needed.

Reply:
I did a test with a minimal example and it works, and without the `ActiveSupport::Concern`:

```ruby
module MobileWorkflow
  module Displayable
    module Steps
      module Form
        def mw_form_section(label:, identifier:)
          raise 'Missing label' if label.nil?
          raise 'Missing identifier' if identifier.nil?

          { item_type: :section, label: label, identifier: identifier }
        end
      end
    end
  end
end

module MobileWorkflow
  module Displayable
    def self.included(base)
      base.extend(Steps::Form)
    end
  end
end

class Example
  include MobileWorkflow::Displayable
end

Example.mw_form_section(label: 'my_label', identifier: 'my ID')
# => {:item_type=>:section, :label=>"my_label", :identifier=>"my ID"}
```

So I think the `extend ActiveSupport::Concern` is not particularly needed in this case and it could be removed

Reply:
Thanks @pjmartorell , it works for me as well.